x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -245,6 +267,7 @@ public static String documentId(String id) {
         private List<DataFrameAnalysisConfig> analyses;
         private Map<String, Object> query = Collections.singletonMap(MatchAllQueryBuilder.NAME, Collections.emptyMap());
         private FetchSourceContext analysesFields;
+        private ByteSizeValue modelMemoryLimit;
```

Comment:
Should we make the default value visible? If so we don't need to deal with `null`s in the code plus we communicate better to new users that there is such a setting.

Reply:
It's tricky because the behaviour is that an explicit setting lower than the max is an error but a default lower than the max is silently capped.  So we need to remember the difference between a value the user set and one from the code.  `AnalysisLimits` allows `null` for the same reason.